File: 2303. 计算应缴税款总额/2303. 计算应缴税款总额.java
class Solution {
    public double calculateTax(int[][] brackets, int income) {
        int tax = 0;
        int last = 0;
        for (int i = 0; i < brackets.length; i++) {
            if (brackets[i][0] <= income) {
                tax += (brackets[i][0] - last) * brackets[i][1];
                last = brackets[i][0];
            } else {
                tax += (income - last) * brackets[i][1];
                break;
            }
        }

        return (double) tax / 100;
    }
}

File: 2303. 计算应缴税款总额/2303. 计算应缴税款总额.py
class Solution:
    def calculateTax(self, brackets: List[List[int]], income: int) -> float:
        tax = 0
        last = 0
        for b in brackets:
            if b[0]<=income:
                tax += b[1]*(b[0]-last)/100
                last = b[0]
            else:
                tax += b[1]*(income-last)/100
                return tax
        return tax